### ISSUE TITLE
DataStore not syncing

### ISSUE THREAD

**Issue by calvinmuller** [aws-amplify/amplify-flutter#936]
**Describe the bug**
It seems when using @auth() the subscriptions fail to connect and then send the subsequent updates.

**To Reproduce**
1. `schema.json`
```
type Todo @model @auth(rules: [{allow: owner, operations: [create, delete, update, read]}]) {
  id: ID!
  name: String!
  description: String
}

type Boards @model @auth(rules: [{ allow: owner }]) {
  id: ID!
  name: String!
  description: String
  length: String
  width: String
  manufacturer: String
  volume: String
  images: [String]
}
```
2.  amplify models codegen
3. Start up app (already logged in) 

**Expected behavior**
It should sync and not show odd errors in the console

**Platform**
Amplify Flutter current supports iOS and Android.  This issue is reproducible in (check all that apply):
[] Android
[x] iOS

** Flutter **
```
[✓] Flutter (Channel stable, 2.5.1, on macOS 11.2.3 20D91 darwin-x64, locale en-ZA)
    • Flutter version 2.5.1 at /usr/local/Caskroom/flutter/2.2.3/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ffb2ecea52 (2 weeks ago), 2021-09-17 15:26:33 -0400
    • Engine revision b3af521a05
    • Dart version 2.14.2

```

**Smartphone (please complete the following information):**
 - Device: iPhone 11
 - Os: iOS14.6

**Additional context**
I get this when launching my app and being signed in.
```

[StarscreamAdapter] socket.write - {"id":"3FB782F7-08D2-4F56-BAA1-5B69C38423E4","payload":{"data":"{\"query\":\"subscription OnCreateTodo($owner: String!) {\\n  onCreateTodo(owner: $owner) {\\n    id\\n    description\\n    name\\n    __typename\\n    _version\\n    _deleted\\n    _lastChangedAt\\n    owner\\n  }\\n}\",\"variables\":{\"owner\":\"\"}}","extensions":{"authorization":{"Authorization":"MYJWTTOKEN.IS.HERE","host":"xxx.appsync-api.eu-west-1.amazonaws.com"}}},"type":"start"}

[StarscreamAdapter] websocketDidReceiveMessage: - {"id":"3FB782F7-08D2-4F56-BAA1-5B69C38423E4","type":"error","payload":{"errors":[{"errorType":"Unauthorized","message":"Not Authorized to access onCreateTodo on type Subscription"}]}}
```

When saving a model eg.
```
Successfully saved model: FlutterSerializedModel(id: "808b0852-932d-445a-af0e-e0633e938bde", values: ["description": Amplify.JSONValue.null, "id": Amplify.JSONValue.string("808b0852-932d-445a-af0e-e0633e938bde"), "volume": Amplify.JSONValue.string("60"), "manufacturer": Amplify.JSONValue.string("Swordfish"), "length": Amplify.JSONValue.string("9\'4"), "name": Amplify.JSONValue.string("Cheddar Cheese"), "images": Amplify.JSONValue.array([Amplify.JSONValue.string("image_picker_8CECB76A-E7B2-492D-A87C-0C4206C6389C-1587-000000EBAB95846D.jpg")]), "width": Amplify.JSONValue.null])

Failed to parse the event DataStoreError: No decoder found for model named MutationEvent
Recovery suggestion: There is no decoder registered for the model named MutationEvent. Register models with `ModelRegistry.register(modelName:)` at startup.
```

**Comment by HuiSF** [comment]
Hi @calvinmuller thanks for opening this issue. What auto provider do you use for your project please?

**Comment by calvinmuller** [comment]
@HuiSF Thanks for the repy, I'm using Cognito userpools.

**Comment by LetsHandOutOurMastery** [comment]
For me also its not working. Whenever Im trying to save data in aws amplify datastore, im receiving this exception.

DataStoreException{message=Failure during mutation., cause=ApiException{message=OkHttp client request failed., cause=javax.net.ssl.SSLHandshakeException: Chain validation failed, recoverySuggestion=See attached exception for more details.}, recoverySuggestion=Check details.}
W/amplify:aws-datastore(31591): at com.amplifyframework.datastore.appsync.AppSyncClient.lambda$mutation$5(AppSyncClient.java:359)
W/amplify:aws-datastore(31591): at com.amplifyframework.datastore.appsync.-$$Lambda$AppSyncClient$FfgHx1XvivjmTcZhtf9BjbrAq_M.accept(Unknown Source:4)
W/amplify:aws-datastore(31591): at com.amplifyframework.api.aws.AppSyncGraphQLOperation$OkHttpCallback.onFailure(AppSyncGraphQLOperation.java:157)
W/amplify:aws-datastore(31591): at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:525)
W/amplify:aws-datastore(31591): at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
W/amplify:aws-datastore(31591): at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
W/amplify:aws-datastore(31591): at java.lang.Thread.run(Thread.java:920)
W/amplify:aws-datastore(31591): Caused by: ApiException{message=OkHttp client request failed., cause=javax.net.ssl.SSLHandshakeException: Chain validation failed, recoverySuggestion=See attached exception for more details.}
W/amplify:aws-datastore(31591): ... 5 more
W/amplify:aws-datastore(31591): Caused by: javax.net.ssl.SSLHandshakeException: Chain validation failed

**Comment by HuiSF** [comment]
@LetsHandOutOurMastery Thanks for following up. Though I don't think the exception `javax.net.ssl.SSLHandshakeException: Chain validation failed` is related to this thread or amplify plugins...
This [stack overflow question](https://stackoverflow.com/questions/45923747/android-emulator-chain-validation-failed-connecting-developers-machine-with-se) may provide some hits how to resolve.

**Comment by HuiSF** [comment]
Seems something went wrong with [this commit](https://github.com/aws-amplify/amplify-flutter/commit/1d27acbdf0c44e60e1efcc23bc0c9ae9d9adf139).

**Comment by HuiSF** [comment]
It looks like above linked PR won't fix this issue.

As [this comment](https://github.com/aws-amplify/amplify-ios/issues/1350#issuecomment-899061425) stated, in the iOS native models, it's should contain `provider` info to indicate which auth provider to use. This is a breaking change for amplify-flutter which requires code change on [converting serialized schema to native schema](https://github.com/aws-amplify/amplify-flutter/blob/main/packages/amplify_datastore/ios/Classes/types/model/FlutterAuthRule.swift#L85-L94).

This would firstly require amplify-codegen changes to insert correct auth provider info to generate dart models.

**Comment by calvinmuller** [comment]
hi @HuiSF thank you so much for your efforts. so if i drop from 0.2.4 to 0.2.2 data store and flutter package i'll be able to bypass this? thanks again!

**Comment by HuiSF** [comment]
Yes @calvinmuller 0.2.2 model subscription with auth owner rule should work.

**Comment by HuiSF** [comment]
Hello @calvinmuller the new version [0.2.5](https://github.com/aws-amplify/amplify-flutter/releases/tag/v0.2.5) has been released which includes the fix for the model subscription issue happening in iOS platform. Please upgrade and test it out, and feel free to follow up if you encounter any issues! Thanks for the feedback :)

**Comment by martinzuccotti** [comment]
Hi, I'm getting the same error as reported here with version `0.3.2`

Error message:
```
Failed to parse the event DataStoreError: No decoder found for model named MutationEvent
Recovery suggestion: There is no decoder registered for the model named MutationEvent. Register models with `ModelRegistry.register(modelName:)` at startup.
Encoder Foundation.(unknown context at $111268be8).__JSONEncoder
```

Flutter:
```
[✓] Flutter (Channel stable, 2.10.2, on macOS 12.2.1 21D62 darwin-arm, locale en-AR)
    • Flutter version 2.10.2 at ~/fvm/versions/stable
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 097d3313d8 (12 days ago), 2022-02-18 19:33:08 -0600
    • Engine revision a83ed0e5e3
    • Dart version 2.16.1
    • DevTools version 2.9.2
```

**Comment by HuiSF** [comment]
Hi @martinzuccotti you can safely ignore this error.
Thanks!

**Comment by wolfeiq** [comment]
Hey, I'm getting this error with amplify_flutter: ^0.6.0. How do I fix this? Only on iOS

**Comment by wolfeiq** [comment]
> Hi @martinzuccotti you can safely ignore this error.
> Thanks!

New versions of amplify_flutter do not provide solutions for this, and if I ignore it, the app doesn't sync with datastore. Pls help! :(
